components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -119,15 +114,32 @@ impl CanvasContextState {
     }
 }
 
+#[must_root]
 #[derive(JSTraceable, MallocSizeOf)]
 pub struct CanvasState {
     #[ignore_malloc_size_of = "Defined in ipc-channel"]
     ipc_renderer: IpcSender<CanvasMsg>,
     canvas_id: CanvasId,
+    state: DomRefCell<CanvasContextState>,
```

Comment:
We no longer need DomRefCell or Cell in members of CanvasState, since CanvasState is stored in a DomRefCell. We can write methods that require `&mut self` instead of `&self` when we need to mutate members now.